src/config/source.rs

```diff
@@ -29,9 +29,14 @@ impl Source {
   }
 
   pub fn detect(name: &str) -> Result<Source, UnsupportedFileFormat> {
-    ALL
-      .into_iter()
-      .find(|format| format.ends_with(name))
-      .ok_or_else(|| UnsupportedFileFormat(name.to_string()))
+    let re = regex::Regex::new(r"\.(graphql|yml|yaml|json)$").unwrap();
```

Comment:
why introduce regex here? looks like we still doing the same iteration later

Reply:
@tusharmath suggested me to do the same